jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStep.java

```diff
@@ -140,40 +143,43 @@ public ImagesAndRegistryClient call()
             LogEvent.lifecycle(
                 "The base image requires auth. Trying again for " + imageReference + "..."));
 
-        Credential registryCredential =
+        Credential credential =
```

Comment:
It seems a little strange that we have so much auth code in PullBaseImage step and not somewhere else like RegistryAuthenticator?

Reply:
(First of all, I don't know the initial intention of `RegistryAuthenticator`, but it has always been "BearerAuthenticator" that's all about bearer auth.)

As discussed offline, we could have introduced an overarching `RegistryClient.doRightAuth()` that perhaps does
```java
if (!doPull/PushBearerAuth()) {
 ... 
 configureBasicAuth();
}
```
I expect this flow should work for both pull and push auth with any registries, and this is exactly the flow we do for target registry auth. However, were taking a different flow for base image registry (as can be seen in the PR), and what is one of the reasons we don't have `doRightAuth()`.

Given this and that `RegistryClient` is a low-level API, it also seemed reasonable to have separate nobs for configuring basic auth and completing bearer auth. Another factor to this is the different between configuring basic auth and completing the whole bearer auth leg: configuring basic auth means that you'll just send the credentials going forward no matter what, while completing bearer auth means interacting with a registry server and an auth server with two round-trips to get another form of credentials to send going forward.